File: gym/aizu/ALDS1_6_A/Main.java
import java.util.*;

public class Main {
  public static void main(String[] args) {
    try (Scanner in = new Scanner(System.in)) {
      int n = in.nextInt();
      int[] a = new int[n];

      for (int i = 0; i < n; i++) {
        a[i] = in.nextInt();
      }

      sort(a, n);
      display(a, n);
    }
  }

  private static void sort(int[] a, int n) {
    int mn = a[0], mx = a[0];

    for (int i = 1; i < n; i++) {
      mn = Math.min(mn, a[i]);
      mx = Math.max(mx, a[i]);
    }

    int k = mx - mn + 1;
    int[] freq = new int[k];

    for (int e : a) {
      freq[e - mn]++;
    }

    int idx = 0, elem = 0;
    for (int f : freq) {
      while (f-- > 0) {
        a[idx++] = elem + mn;
      }
      elem++;
    }
  }

  private static void display(int[] a, int n) {
    for (int i = 0; i < n; i++) {
      System.out.printf("%d", a[i]);
      if (i < n - 1) {
        System.out.print(" ");
      }
    }
    System.out.println();
  }
}

File: gym/aizu/ALDS1_6_A/main.rs
use std::io;

fn get_num() -> usize {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    input.trim().parse::<_>().unwrap()
}

fn get_nums() -> Vec<i32> {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();

    input
        .trim()
        .split_whitespace()
        .map(|d| d.parse::<i32>().unwrap())
        .collect::<Vec<_>>()
}

fn sort(a: &mut [i32], mn: i32, mx: i32) {
    let k = (mx - mn + 1) as usize;
    let mut freq = vec![0; k];

    for e in a.iter() {
        freq[(*e - mn) as usize] += 1;
    }

    let mut idx = 0;
    let mut elem = 0;

    for f in &freq {
        for _ in 0..*f {
            a[idx] = elem + mn;
            idx += 1;
        }
        elem += 1;
    }
}

fn display(a: &[i32], n: usize) {
    for i in 0..n - 1 {
        print!("{} ", a[i]);
    }
    println!("{}", a[n - 1]);
}

fn main() {
    let n = get_num();
    let mut a = get_nums();

    let mn = {
        let tmp_mn = a.iter_mut().min().unwrap();
        *tmp_mn
    };

    let mx = {
        let tmp_mx = a.iter_mut().max().unwrap();
        *tmp_mx
    };

    sort(&mut a, mn, mx);

    display(&a, n);
}